torch/_inductor/cpp_builder.py

```diff
@@ -116,9 +117,25 @@ def install_gcc_via_conda() -> str:
     return cxx_path
 
 
+@functools.lru_cache(None)
+def check_compiler_exist_windows(compiler: str) -> None:
+    """
+    Check if compiler is ready, in case end user not activate MSVC environment.
+    """
+    try:
+        output_msg = (
+            subprocess.check_output([compiler, "/help"], stderr=subprocess.STDOUT)
+            .strip()
+            .decode(*SUBPROCESS_DECODE_ARGS)
+        )
+    except FileNotFoundError as exc:
+        raise RuntimeError(f"Compiler: {compiler} is not found.") from exc
+
+
 def get_cpp_compiler() -> str:
     if _IS_WINDOWS:
         compiler = os.environ.get("CXX", "cl")
+        check_compiler_exist_windows(compiler)
```

Comment:
This compiler availability check and informative warning message looks good but it doesn't seem to be windows-specific? Other platforms like Linux/Mac also need it?

Reply:
@jgong5 already have code to check compiler exist for Non-Windows OS.
<img width="931" alt="image" src="https://github.com/user-attachments/assets/311faf4f-dcf2-4059-b7ae-6866aec7a17c">